### PR TITLE
Feature: Create donation flow (kids-1421)

### PR DESCRIPTION
**Pull request by Daniela510**
## Release Notes

- **New Features**
	- Added a new event type `parentReflectionFlowOrganisationClicked` for enhanced analytics tracking.
	- Introduced streamlined access to `GiveBloc` and `MediumCubit` through dependency injection, improving code maintainability.
	- Enhanced `GratefulScreen` functionality with improved state handling and navigation for donation flows.

- **Bug Fixes**
	- Simplified profile filtering in `GratefulCubit` to retain all children regardless of gratitude status.

- **Refactor**
	- Refactored state management in various pages to use dependency injection for better clarity and maintainability.

### CODE REVIEW

File: lib/features/family/app/injection.dart [hunk 1]
```diff
@@ -53,6 +55,15 @@ void initCubits() {
     ..registerLazySingleton<CameraCubit>(
       CameraCubit.new,
     )
+    ..registerLazySingleton<MediumCubit>(MediumCubit.new)
+    ..registerLazySingleton<GiveBloc>(
```
Comment (TammiLion) [inline lib/features/family/app/injection.dart:59]: Nice work!

File: lib/features/family/features/reflect/presentation/pages/grateful_screen.dart [hunk 2]
```diff
@@ -42,66 +58,157 @@ class _GratefulScreenState extends State<GratefulScreen> {
...
+        ),
+      ).toRoute(context),
+    );
+  }
+
+  Future<void> _navigateToParentGivingScreen(
+    BuildContext context,
+    Organisation org,
+  ) async {
+    _medium.setMediumId(org.namespace);
```
Comment (TammiLion) [inline lib/features/family/features/reflect/presentation/pages/grateful_screen.dart:166]: Do we need
`    await context.read<ProfilesCubit>().setActiveProfile(profile.userId);`
here?

Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/pages/grateful_screen.dart:166]: no, thats for kids

Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/pages/grateful_screen.dart:166]: the GiveBloc uses the userid from the AuthCubit, not ProfilesCubit

Comment (TammiLion) [reply lib/features/family/features/reflect/presentation/pages/grateful_screen.dart:167]: Clear, ty

**Review by TammiLion — APPROVED** [review]
just the google meet discussion thingie, but all else approved!

**Pull request merged**